### ISSUE TITLE
Cogniterra

### ISSUE THREAD

**Issue by OrifjonOlimjonov** [RedJocker/BlackboardTestCreation#1]
Is there the Cogniterra editing link?

**Comment by RedJocker** [comment]
no, not yet

**Comment by RedJocker** [comment]
created cogniterra link and added images for first 2 stages.
invitation link is on trello card for the project.
But for stage descriptions changes make a comment about the desired change in some issue and make a pull request for it here.
Don't just change there because I'll be doing copy and paste from all source code for a given stage description from github to cogniterra and possibly overwrite some edit if you just edit on cogniterra. Also github stage descriptions should be always on most recent version.

**Comment by OrifjonOlimjonov** [comment]
> created cogniterra link and added images for first 2 stages. invitation link is on trello card for the project. But for stage descriptions changes make a comment about the desired change in some issue and make a pull request for it here. Don't just change there because I'll be doing copy and paste from all source code for a given stage description from github to cogniterra and possibly overwrite some edit if you just edit on cogniterra. Also github stage descriptions should be always on most recent version.

should I fork it? or is it possible to send a pull request even if I have cloned it?

**Comment by RedJocker** [comment]
I usually do by forking working on my fork and making the pull request through github. I guess there are other ways, but I know just the basics to make things work.

**Comment by OrifjonOlimjonov** [comment]
I think I need to install a program called Mockoon.

**Comment by RedJocker** [comment]
> I think I need to install a program called Mockoon.

Yes. Users can have the option to make a backend themselves to do manual testing, but I'm predicting that almost every one will prefer to use something that was prepared by us for that and the way I found that was easier to do that was using this program.
Since you are revising the project it is expected that you assert that the mockoon setup is also working and behaving as expected to make manual testing possible for users.
Our tests do not depend on that, it is just a convenience for users, but they will most probably be using it.

**Comment by RedJocker** [comment]
I have added short descriptions for all stages.
I'm still unsure if pre-requisite sections is complete, if you see something that could be added that is not already an implicit pre-requisite though dependency chains please bring it up.

**Comment by RedJocker** [comment]
@OrifjonOlimjonov added images to stage3

**Comment by RedJocker** [comment]
added images to stage4 and stage5.
I believe creation phase is done, but you may propose changes yet in your review.
Please also make sure tests are ok and not missing anything essential.
Besides that is just preparing for publishing.

**Comment by OrifjonOlimjonov** [comment]
https://github.com/RedJocker/BlackboardTestCreation/assets/98660391/a7551d04-22d2-495c-bc1d-67ab0efbbd01

I think we should add a test for this case.

**Comment by RedJocker** [comment]
There are several cases for invalid login already, I don't see a special reason to add a test case for untrimmed username. I think that it is even better to leave that as an implementation choice, I didn't allow extra whitespaces on end, but it is sort of annoying because the autocomplete does put an extra whitespace, so maybe an implementation that trims whitespaces would actually be a good idea.

**Comment by OrifjonOlimjonov** [comment]
I have updated Prerequisites. Please check it out. maybe i added an unnecessary topic. I think this project type of Challenging. That's why I didn't add basic UI components.

**Comment by RedJocker** [comment]
Thank you very much for the review, I asked alexadra for this review not only because I think it is a good practice, but also I was thinking this would be a good opportunity for you to improve your skills in writing tests for android track so that you can write better projects and with more autonomy. I know we were rushing a bit and that there were some pressure for you to finish review fast, but I hope it was a good experience anyway.
Keep up the good work, hope to collaborate with you in other projects.
